node/pkg/node/node.go

```diff
@@ -346,12 +357,12 @@ func GuardianOptionAdminService(socketPath string, ethRpc *string, ethContract *
 func GuardianOptionPublicRpcSocket(publicGRPCSocketPath string, publicRpcLogDetail common.GrpcLogDetail) *GuardianOption {
 	return &GuardianOption{
 		name:         "publicrpcsocket",
-		dependencies: []string{"governor"},
+		dependencies: []string{"governor", "db"},
 		f: func(ctx context.Context, logger *zap.Logger, g *G) error {
 			// local public grpc service socket
 			publicrpcUnixService, publicrpcServer, err := publicrpcUnixServiceRunnable(logger, publicGRPCSocketPath, publicRpcLogDetail, g.db, g.gst, g.gov)
 			if err != nil {
-				logger.Fatal("failed to create publicrpc service socket", zap.Error(err))
+				return err
```

Comment:
Same as before. The context is useful and makes it easier to track down where the error comes from.